subcommands/version/updates.go

```diff
@@ -0,0 +1,145 @@
+package version
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/hashicorp/go-version"
+	tuf "github.com/theupdateframework/go-tuf/client"
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+type FioctlUpdateFinder struct {
+	tuf        *tuf.Client
+	localStore *jsonFilesStore
+	platform   string
+}
+
+type FioctlUpdate struct {
+	Version *version.Version
+	Uri     string
+	Sha512  data.HexBytes
+	len     int64
+}
+
+func NewFioctlUpdateFinder(localJsonPath string) (*FioctlUpdateFinder, error) {
+	remote, err := tuf.HTTPRemoteStore(RootUrl, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	local := &jsonFilesStore{localJsonPath}
+	tuf := tuf.NewClient(local, remote)
+
+	platform := runtime.GOOS + "/" + runtime.GOARCH
+	return &FioctlUpdateFinder{tuf, local, platform}, nil
+}
+
+type fioctlCustom struct {
+	Platform string `json:"platform"`
+	Uri      string `json:"uri"`
+	Version  string `json:"version"`
+}
+
+func (f *FioctlUpdateFinder) FindLatest() (*FioctlUpdate, error) {
+	latestVer, err := version.NewVersion(Commit)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse binary's version: %s", err)
+	}
+
+	if !f.localStore.initialized() {
+		fmt.Println("Initializing local TUF storage")
+		if err := f.tuf.Init(RootContent); err != nil {
+			return nil, err
+		}
+	}
+	if _, err = f.tuf.Update(); err != nil {
+		return nil, err
+	}
+	targets, err := f.tuf.Targets()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Does it work well if there are no targets yet, or they are empty?

Not sure this is of any criticality. I was just thinking about the _first release with TUF_.

Reply:
yes. its okay. with the caveat below to mike's comment.